src/deb/control/postinst

```diff
@@ -1,17 +1,87 @@
-#!/bin/bash
+#!/usr/bin/env bash
+
+#set -x
 set -o errexit
 
-echo 'postinst script executing'
+. /usr/share/debconf/confmodule
+
+NAME=ned
+
+TOMCAT7_USER=$NAME
+TOMCAT7_GROUP=$NAME
+HOME=/opt/plos/$NAME
+
+function process_env_template {
+	CONTEXT_TEMPALTE=$1
+
+  ls -lh $CONTEXT_TEMPALTE
+
+	echo "Processing template $CONTEXT_TEMPALTE"
+
+  eval "cat <<EOF
+$(<$CONTEXT_TEMPALTE)
+EOF
+" > $CONTEXT_TEMPALTE
+}
+
+case "$1" in
+    configure)
+
+  # create user/group
+
+	if ! getent group "$TOMCAT7_GROUP" > /dev/null 2>&1 ; then
+	    addgroup --system "$TOMCAT7_GROUP" --quiet
+	fi
+	if ! id $TOMCAT7_USER > /dev/null 2>&1 ; then
+	    adduser --system --home $HOME --no-create-home \
+		--ingroup "$TOMCAT7_GROUP" --disabled-password --shell /bin/false \
+		"$TOMCAT7_USER"
+	fi
 
-#TODO: decouple this from salt?
-echo "source /etc/ned/deploy-envs.sh"
-source /etc/ned/deploy-envs.sh
+  chmod +x $HOME/bin/*.sh
+  chown -Rh $TOMCAT7_USER:$TOMCAT7_GROUP $HOME
 
-echo "bash /usr/share/nedinstall/bin/install.sh"
-bash /usr/share/nedinstall/bin/install.sh
+  GET="db_get [[artifactId]]"
+
+  # Fetching configuration from debconf
+
+  $GET/ned_db_host && export NED_DB_HOST=$RET
+  $GET/ned_db_user && export NED_DB_USER=$RET
+  $GET/ned_db_password && export NED_DB_PASSWORD=$RET
+  $GET/ringgold_db_host && export RINGGOLD_DB_HOST=$RET
+  $GET/ringgold_db_user && export RINGGOLD_DB_USER=$RET
+  $GET/ringgold_db_password && export RINGGOLD_DB_PASSWORD=$RET
+  $GET/ringgold_import_dir && export RINGGOLD_DIR=$RET
+  $GET/http_port && export HTTP_PORT=$RET
+  $GET/control_port && export CONTROL_PORT=$RET
+
+  # removes the saved values, so they need to be re-entered the next time around
+  db_purge
+
+  # generate configs from templates
+  process_env_template $HOME/conf/context.xml
+  cat $HOME/conf/context.xml
+
+  process_env_template $HOME/conf/server.xml
+#  cat $HOME/conf/server.xml
+
+  # run the migrations and import ringgold
+  export MIGRATIONS_DIR=$HOME/migrations
+  $HOME/bin/migrate.sh
+
+esac
+
+# In case this system is running systemd, we make systemd reload the unit files
+# to pick up changes.
+if [ -d /run/systemd/system ] ; then
+	systemctl --system daemon-reload >/dev/null || true
+fi
 
-# start ned service, if exists.    
-if service --status-all 2>&1 | grep -wq 'ned'; then    
-  echo 'sudo service ned start'
-  sudo service ned start 
+# Automatically added by dh_installinit
+if [ -x "/etc/init.d/$NAME" ] || [ -e "/etc/init/$NAME.conf" ]; then
+	if [ ! -e "/etc/init/$NAME.conf" ]; then
+		update-rc.d $NAME defaults 92 08 >/dev/null
+	fi
+	invoke-rc.d $NAME start || true
 fi
+# End automatically added section
```

Comment:
It would be preferable to use the standard dh_make skeleton here, but since you have isolated the "configure" action this should technically work alright.